### DISCUSSION TITLE
Hover Scrubbing and Quick Selection (like in FCX)

### DISCUSSION THREAD

**Discussion by anciltech** [mifi/lossless-cut#1419] [Ideas]
Hi there!
The usability of losslesscut would increase 10x if **Hover Scrubbing** and **Click and Drag to Select** were implemented in a similar way to FCX or other video editing software (see attached Video).

This could speed up the cutting process _a lot_ and make the software more enjoyable to use. 
![FCX Example](https://user-images.githubusercontent.com/79978026/212156577-de0b9eaf-9a11-45f3-a5a2-19afe8140013.gif)

**Comment by mifi** [comment]
Hi!
You mean hover scrubbing on the main timeline or on segments in the segment panel ,or something else?

And for drag to select, what is the selection used for exactly?

**Comment by anciltech** [comment]
Hi @mifi, 

> "hover scrubbing on the main timeline" 


     Yes exactly, at least on the main timeline. Of course performance is key here, there would have to be minimal lag between change of position over the filmstrip and the preview which is shown. 

> drag to select, what is the selection used for exactly?


     The selections would define your "Segments". 
So if I: 

- hover across a filmstrip and see an area i want to keep/remove
- I would click and hold at the Start/End of what I want to keep (Segment Start time is then defined)
- and then (while still holding the click) drag across the filmstrip and release once I find my Start/End point (Segment End time is then defined)

What would really turn it up to 11 is if you could have filmstrips for multiple files so the user could quickly trim down multiple clips. 
Essentially what I am picturing is exactly what you see in Final Cut Pro, where you see multiple Filmstrips representing your video files, yellow boxes around your selections, and a larger Preview window which shows what you are Hover Scrubbing across.
The user could quickly Hover Scrub across these filmstrips and Click Drag to select the Segments to Keep/Remove, and then hit a confirm button to verify there selection and carry out the cutting process. 

What are your thoughts on this? I can work on defining the concept more thoroughly if this is of interest to you. 

Cheers
Max

**Reply by mifi** [reply]
as for Hover Scrubbing, I agree it's a good thing, and it has been suggested here: https://github.com/mifi/lossless-cut/issues/483

As for Click and drag to Select, we could do that on the batch file list (left side) - if also showing previews/thumbnails of those files in the left side panel. I think it's an interesting idea, but I's not exactly compatible how LosslessCut works right now, because in order to define segments for a file in the batch list, the file has to have first been opened up. Maybe we could open up the file on MouseDown, though, then on mouseUp, add the segment. However if the file didn't open in time, then it complicates a bit. It ithink it's tricky to implement correctly but might be possible. Will add the suggestion to #483

**Reply by anciltech** [reply]
Ah yeah, I never used the batch feature, that seems like the perfect place for these filmstrip previews.

hm ok, by "open up", does that mean that the file is loaded into ram, or what exactly happens at that point? I'm not sure how other programs achieve video playback without specifically opening the file, but if there is a preview built for the Hover Scrubbing than maybe that would be enough to just quickly select segments. In my opinion it wouldnt have to be high detail to be able to tell if a user wants to keep the section or not.
